mtp_api/apps/transaction/tests/test_bank_admin_views.py

```diff
@@ -82,3 +85,294 @@ def test_create_list(self):
             ).count(),
             len(data_list)
         )
+
+    def _test_create_list_incomplete_data(self, null_field):
+        url = self._get_url()
+        data_list = self._get_transactions_data()
+
+        user = self.bank_admins[0]
+        data_list[0][null_field] = None
+
+        current_count = Transaction.objects.count()
+
+        response = self.client.post(
+            url, data=data_list, format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(user)
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_400_BAD_REQUEST)
+
+        # check no change in db
+        self.assertEqual(current_count, Transaction.objects.count())
+
+    def test_create_list_null_account_number_bad_request(self):
+        self._test_create_list_incomplete_data('sender_account_number')
+
+    def test_create_list_null_sort_code_bad_request(self):
+        self._test_create_list_incomplete_data('sender_sort_code')
+
+    def test_create_list_null_amount(self):
+        self._test_create_list_incomplete_data('amount')
```

Comment:
I don't think they "happen" to have the same result - the test is based on recognising that a required field is absent, and those are all required fields.

I don't really like the linked dataprovider thing. I think adding syntactic sugar for calling a function multiple times with different values is overkill, and it also prevents you from naming tests according to the rules that you have been espousing i.e. specifying inputs in the method name.

How about if I move just the assertions out of the base function and into the callers? That way if the expected per-field behaviour changes it's easy to alter, but it doesn't duplicate the whole "setting a field to null and making a create request with it" thing which is unlikely to differ?